packages/components/src/components/date-picker/date-picker.tsx

```diff
@@ -158,16 +179,19 @@ const DatePicker = React.memo(props => {
      */
     // onClickClear = () => {};
 
-    const getTargetValue = (new_date, new_duration) => {
+    const getTargetValue = (new_date: string | null, new_duration: string | number | null) => {
         const calendar_value = getCalendarValue(new_date) && toMoment(getCalendarValue(new_date));
         return mode === 'duration' ? new_duration : calendar_value;
     };
 
-    const getInputValue = () => (mode === 'duration' ? duration : date);
+    const getInputValue = (): string | number => (mode === 'duration' ? (duration ? duration : 0) : date);
 
-    const getCalendarValue = new_date => {
+    const getCalendarValue = ((new_date: string | null): string | null => {
         if (!new_date) return isMobile() ? null : toMoment(start_date || max_date).format(date_format);
         return convertDateFormat(new_date, display_format, date_format);
+    }) as {
+        (new_date: string): string;
+        (new_date: string | null): string | null;
     };
```

Comment:
@amir-deriv Avoid using `as` 🙏🏻 
As @maryia-deriv mentioned before, You need to make the `value` type optional inside the `Native` component 🙇🏻

Reply:
I have added this overloading to the function `getCalendarValue` , making the `value` optional inside the `Native` component gives couple of type errors on `Native` component, since `value?: string` will take `string` or `undefined` but not `null` 

If we see `getCalendarValue`, passing a valid string will always give as string, so I added these overload

Reply:
removed the `as` here and checked for the `null` condition in `Native` component